src/main/kotlin/no/skatteetaten/aurora/mokey/service/ApplicationDataService.kt

```diff
@@ -42,11 +46,45 @@ class ApplicationDataService(
         return cache[id]?.publicData
     }
 
-    fun findPublicApplicationDataByApplicationDeploymentRef(applicationDeploymentRefs: List<ApplicationDeploymentRef>): List<ApplicationPublicData> {
-        return cache.filter {
+    fun findPublicApplicationDataByApplicationDeploymentRef(
+        applicationDeploymentRefs: List<ApplicationDeploymentRef>,
+        cached: Boolean = true
+    ): List<ApplicationPublicData> {
+        val cachedElements = cache.filter {
             val publicData = it.value.publicData
-            applicationDeploymentRefs.contains(ApplicationDeploymentRef(publicData.environment, publicData.applicationDeploymentName))
-        }.ifEmpty { return emptyList() }.values.map { it.publicData }
+            applicationDeploymentRefs.contains(ApplicationDeploymentRef(
+                publicData.environment,
+                publicData.applicationDeploymentName
+            ))
+        }.values
+
+        return when (cached) {
+            true -> cachedElements.map { it.publicData }
+            false -> {
```

Comment:
Når man henter ferske data så burde vel cache oppdateres med disse, slik at man ikke får divergerende visninger i andre applikasjoner som skal vise samme info?

Reply:
ja, det må bli slik det er nå. konsollet har allerede en funksjonalitet for å oppdatere cachen, når noe endrer seg feks ved en deploy. Men da blir alt oppdatert.